Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +31,61 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return totalAmpHoursUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    return (currentNow_ < 0);
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+   double hoursTillChargedOrDepleted, msTillChargedorDepleted;
+   int hours, minutes, seconds, milliseconds;
```

Comment:
These variables don't get used until a bit later, it's usually better practice to minimise the time between the time of declaration & time of use when possible. Put these declarations closer to the use.

In a lot of the cases, you can just declare & set the variables in one line, ie instead of
```
int hours;
...
hours = floor(hoursTillChargedOrDepleted);
```
Do
```
int hours = floor(hoursTillChargedOrDepleted);
```